### PR TITLE
[State transitions] Ocrvs 5791 archive

### PR DESCRIPTION
**Pull request by Nil20**

### CODE REVIEW

File: packages/workflow/src/records/handler/archive.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    record,
+    await getLoggedInPractitionerResource(token),
+    reason,
+    comment,
+    duplicateTrackingId
+  )
+
+  await sendBundleToHearth(archivedRecordWithTaskOnly)
+  await indexBundle(archivedRecord, token)
+  await indexBundleForAssignment(
```
Comment (Nil20) [inline packages/workflow/src/records/handler/archive.ts:53]: For unassigning we'll have to send the task to `/events/unassigned` but this handler doesn't update the reg-status. So, for now, I've kept it like this. Will have to update the search handler.

Comment (Nil20) [reply packages/workflow/src/records/handler/archive.ts]: `indexBundle` seems to unassign the record itself. Not sure what happened before.

File: packages/gateway/src/features/registration/root-resolvers.ts [hunk 2]
```diff
@@ -146,7 +147,10 @@ export const resolvers: GQLResolver = {
         hasScope(context.headers, 'validate') ||
         hasScope(context.headers, 'declare')
       ) {
-        context.record = await fetchRegistration(id, context.headers)
+        context.record = await fetchRegistrationForAssignment(
```
Comment (Zangetsu101) [inline packages/gateway/src/features/registration/root-resolvers.ts:150]: Not sure if the changed name is appropriate in all the cases

File: packages/workflow/src/records/fhir.ts [hunk 3]
```diff
@@ -428,6 +428,61 @@ export function createValidateTask(
...
+    code: previousTask.code,
+    focus: previousTask.focus,
+    id: previousTask.id,
+    requester: {
+      agent: { reference: `Practitioner/${practitioner.id}` }
+    },
+    identifier: previousTask.identifier,
+    extension,
+    // Reason example - "duplicate"
+    reason: { text: reason ?? '' },
```
Comment (Zangetsu101) [inline packages/workflow/src/records/fhir.ts:467]: I wonder which would be better, keeping the reason with an empty string or not adding in the reason at all in that case

File: packages/workflow/src/records/handler/archive.test.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        return res(ctx.json({}))
+      })
+    )
+
+    // Sends bundle to unassign in elastic search
+    mswServer.use(
+      rest.post('http://localhost:9090/events/unassigned', (_, res, ctx) => {
+        return res(ctx.json({}))
+      })
+    )
```
Comment (Zangetsu101) [inline packages/workflow/src/records/handler/archive.test.ts:68]: Do we need to unassign a record when archiving?

Comment (Nil20) [reply packages/workflow/src/records/handler/archive.test.ts:68]: Nope. Probably did this when I was indexing the bundle to `/unassign/record` route to un-assign a record which I mentioned earlier in the PR.

File: packages/workflow/src/records/handler/archive.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  id: z.string(),
+  reason: z.string().optional(),
+  comment: z.string().optional(),
+  duplicateTrackingId: z.string().optional()
+})
+
+export async function archiveRecordHandler(
+  request: Hapi.Request,
+  _: Hapi.ResponseToolkit
+) {
```
Comment (Zangetsu101) [inline packages/workflow/src/records/handler/archive.ts:31]: Can we use the "createRoute" function here? We could add a optional parameter about including history records when downloading the record within it

**Review by Zangetsu101 — APPROVED** [review]

**Pull request merged**